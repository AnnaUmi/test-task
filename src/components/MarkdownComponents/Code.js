import React from 'react';
import PropTypes from 'prop-types';

import Hint from './Hint.styles';

const Code = props => {
  const { className, children } = props;
  if (className === 'language-hint') {
    return <Hint>{children}</Hint>;
  }
  return <code id="codes" style={{ fontWeight: 600 }} {...props} />;
};

Code.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

Code.defaultProps = {
  children: '',
  className: '',
};

export default Code;
