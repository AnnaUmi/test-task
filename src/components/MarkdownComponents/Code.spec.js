import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Code from './Code';

jest.mock('./Hint.styles', () => 'HintComponent');

describe('Code', () => {
  it('should match snapshot when className is language-hint', () => {
    const tree = renderer.create(<Code className="language-hint" />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot when className is different from language-hint', () => {
    const tree = renderer.create(<Code className="language-js" />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
