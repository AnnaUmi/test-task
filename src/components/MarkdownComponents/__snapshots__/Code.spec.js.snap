// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code should match snapshot when className is different from language-hint 1`] = `
<code
  className="language-js"
  id="codes"
  style={
    Object {
      "fontWeight": 600,
    }
  }
>
  
</code>
`;

exports[`Code should match snapshot when className is language-hint 1`] = `
<HintComponent>
  
</HintComponent>
`;
